src/hooks/useLocalStorage.hook.ts

```diff
@@ -0,0 +1,40 @@
+import { useState } from 'react';
+
+type SerializableValue =
+  | string
+  | number
+  | boolean
+  | null
+  | SerializableObject
+  | SerializableArray;
+
+type SerializableObject = { [key: string]: SerializableValue };
+type SerializableArray = SerializableValue[];
+
+export function useLocalStorage<T extends SerializableValue>(
+  key: string,
+  defaultValue: SerializableValue,
+): [T, (newValue: T) => void] {
+  const [value, setValue] = useState<T>(() => {
+    const savedValue = localStorage.getItem(key);
+
+    if (savedValue === null) {
+      return defaultValue;
+    }
+
+    try {
+      return JSON.parse(savedValue);
+    } catch (error) {
+      localStorage.removeItem(key);
+
+      return defaultValue;
+    }
+  });
+
+  function save(newValue: T) {
```

Comment:
It is important to adhere to one
```suggestion
  const save = (newValue: T) => {
```